src/errors/ElementAlreadyDefinedError.ts

```diff
@@ -0,0 +1,5 @@
+export class ElementAreadyDefinedError extends Error {
+  constructor(public elementID: string) {
```

Comment:
This is a bit of a personal preference, but I think it's generally the pattern in the code - I would suggest making this variable name camel case and just have it as `elementId`.

Reply:
I think it probably depends on which code you're looking at. ;-)  I'm pretty sure I'm guilty of doing it like `lowerPrefixUpperID` elsewhere myself.  Oops.  But yeah, do what @jafeltra says. ;-)